### ISSUE TITLE
BUG: Misaligned margins for tiddler subtitles with a modified date but no modifier

### ISSUE THREAD

**Issue by Jermolene** [TiddlyWiki/TiddlyWiki5#8657]
A tiddler with a modified field but no modifier field is currently displayed as follows:

![image](https://github.com/user-attachments/assets/39a0dcf1-41c4-4ddf-b7f2-76a2be740c38)

Note how the modification date is not in horizontal alignment with the tiddler title.

I think the problem was introduced in 1ca6305acb5ad4624b7253feac2694581b2df01b

**Comment by pmario** [comment]
I think $:/core/ui/ViewTemplate/subtitle/modifier needs an "if"
Using whitespace for formatting does not work out. 

![grafik](https://github.com/user-attachments/assets/5b6a640c-9a39-4dfd-afe9-a9b0d3b7df9e)

**Comment by pmario** [comment]
PR is incoming. Only need to test the new tests

**Comment by Jermolene** [comment]
Hi @pmario I think we need to add some alignment tests that we run on the empty edition. We could check conditions like:

```
document.querySelector(".tc-title").getBoundingClientRect().top === document.querySelector(".tc-site-title").getBoundingClientRect().top
```

**Comment by pmario** [comment]
I had this in mind. The problem is, that &nbsp; was used for alignment. That's easy to test with this code in TW.

```
title: Subtitle/modifier-modified
description: Tiddler ViewTemplate Subtitle. Show modifier and modified in UTC format
type: text/vnd.tiddlywiki-multiple
import: [all[shadows+tiddlers]tag[$:/tags/ViewTemplate/Subtitle]!has[draft.of]] $:/core/ui/ViewTemplate/subtitle/default
tags: $:/tags/wiki-test-spec

title: $:/language/Tiddler/DateFormat

[UTC]DDth MMM YYYY at hh12:0mmam
+
title: Output

\whitespace trim
{{test||$:/core/ui/ViewTemplate/subtitle/default}}
+
title: test
modifier: test-user
modified: 20241003111606000

text
+
title: ExpectedResult

<p><div class=" tc-reveal"><div class="tc-subtitle tc-clearfix"><a class="tc-tiddlylink tc-tiddlylink-missing" href="#test-user">test-user</a>3rd October 2024 at 11:16am</div></div></p>
```

**But I do have a problem** -- It seems that `$:/plugins/tiddlywiki/jasmine/run-wiki-based-tests.js` does not understand the new `import` field. 

Can you help with that one?

**Comment by Jermolene** [comment]
Hi @pmario we should definitely get the wiki-based tests updated to use the "import" field, but I don't see much value in the test you propose because they ignore CSS. These layout issues really need to be checked geometrically in order to have confidence that we don't suffer regressions.

**Comment by pmario** [comment]
This test has value, since it will fail if someone adds some unwanted whitespace, like `&nbsp;` -- So it would have detected the problem, we did introduce without those test. 

Adding a CSS based test can be a **second** step. But I did never manage to write a working playwright test, that can check alignments.

**Comment by pmario** [comment]
There is a second test, which I did not post. It would also would have failed.

**Comment by Jermolene** [comment]
Fixed in #8659

**Comment by pmario** [comment]
@Jermolene  -- This one can be closed again. fixed with:  Fix misaligned subtitle - second try #8672

**Comment by Jermolene** [comment]
Thanks @pmario
